test/vector-tests/SymbolTransaction.vector.ts

```diff
@@ -0,0 +1,185 @@
+    generationHash: string;
+    transactionHash: string;
+    transactionType: number;
+    signedTransactionHash: string;
+    signerPublicKey: string;
+    signerPrivateKey: string;
+    signedPayload: string;
+    transactionsHash: string;
+    cosignatories: [
+        {
+            publicKey: string;
+            privateKey: string;
+        },
+    ];
+}
+
+describe('transaction cosignatures - test vector', () => {
+    tester.run<TransactionCosignatureVectorItem>('test-transaction-cosignatures.json', (item) => {
+        const generationHash = Converter.hexToUint8(item.generationHash);
+        const payload = Converter.hexToUint8(item.payload);
+        const keyPair = new SymbolKeyPair(Key.createFromHex(item.signerPrivateKey));
+
+        // Act + Assert:
+        const transactionHash = SymbolTransactionUtils.calculateTransactionHash(payload, generationHash);
+        expect(Converter.uint8ToHex(SymbolTransactionUtils.calculateAggregateTransactionsHash(payload))).equal(item.transactionsHash);
+        expect(Converter.uint8ToHex(transactionHash)).equal(item.transactionHash);
+
+        function basicTest(cosigners: KeyPair[], expectedPayload?: string) {
+            const signedTransaction = SymbolTransactionUtils.signWithCosigners(keyPair, payload, generationHash, cosigners);
+
+            //Key Pair is fine
+            expect(keyPair.publicKey.toString()).equal(item.signerPublicKey);
+            expect(keyPair.privateKey.toString()).equal(item.signerPrivateKey);
+
+            // Transaction type is the same.
+            expect(SymbolTransactionUtils.getTransactionType(payload)).equal(item.transactionType);
+
+            // Transaction hash is also fine.
+            expect(Converter.uint8ToHex(transactionHash)).equal(item.transactionHash);
+
+            //Signer public key and signature included in the signed payloads (vector and generated)
+            expect(item.signedPayload.indexOf(item.signerPublicKey)).equal(144);
+            const signedPayloadHex = Converter.uint8ToHex(signedTransaction.payload);
+            expect(signedPayloadHex.indexOf(item.signerPublicKey)).equal(144);
+
+            cosigners.forEach((c) => {
+                expect(signedPayloadHex.indexOf(Converter.uint8ToHex(c.publicKey.toBytes()))).gt(-1);
+                const cosignature = c.sign(signedTransaction.transactionHash);
+                expect(signedPayloadHex.indexOf(Converter.uint8ToHex(cosignature))).gt(-1);
+            });
+
+            // Same signed payload!!
+            if (expectedPayload) expect(signedPayloadHex).eq(expectedPayload);
+
+            // Same transaction hash after transaction is signed
+            expect(Converter.uint8ToHex(signedTransaction.transactionHash)).equal(item.signedTransactionHash);
+        }
+        const allCosigners = item.cosignatories.map((pair) => new SymbolKeyPair(Key.createFromHex(pair.privateKey)));
+        basicTest(allCosigners, item.signedPayload.toUpperCase());
+        basicTest([]);
```

Comment:
why passing an empty array?

Reply:
It's testing that if you provide partial cosigners, the transaction hash and other attributes are kept the same. A co-signature doesn't affect the transaction hash like the top level signature does. I'll make this nicer

Reply:
add comments or a separate test perhaps?

Reply:
Comment added